### PR TITLE
Prepare engine for MOJO

### LINKED ISSUE #428: Add gamepad input support

### PR DESCRIPTION
**Pull request by RiscadoA**
Closes #428, #367, #271 and #434.

### CODE REVIEW

File: core/include/cubos/core/gl/debug.hpp [hunk 1]
```diff
@@ -19,6 +19,15 @@ namespace cubos::core::gl
...
         static void init(RenderDevice& renderDevice);
 
+        /// Draw a line that will stay visible for a specified amount of time.
+        /// @param start Starting point of the line.
+        /// @param end Ending point of the line.
+        /// @param relative Whether the ending point is relative to the starting point.
+        /// @param color Color of the drawn line.
+        /// @param time How long should the line be visible for?
+        static void drawLine(glm::vec3 start, glm::vec3 end, bool relative = false, glm::vec3 color = glm::vec3(1),
+                             float time = 0);
```
Comment (RiscadoA) [inline core/include/cubos/core/gl/debug.hpp:29]:
```suggestion
                             float time = 0.0F);
```

File: core/include/cubos/core/io/gamepad.hpp [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        /// @brief Gets the value of an axis.
+        /// @param axis Axis to get the value of.
+        /// @return Value of the axis.
+        float axis(GamepadAxis axis) const
+        {
+            return axes[static_cast<int>(axis)];
+        }
+    };
+
+    /// Convert a GamepadButton enum to a string.
```
Comment (RiscadoA) [inline core/include/cubos/core/io/gamepad.hpp:72]:
```suggestion
    /// @brief Convert a GamepadButton enum to a string.
```

File: core/include/cubos/core/io/gamepad.hpp [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    /// @param key The GamepadButton enum to convert.
+    /// @return The string representation of the GamepadButton enum.
+    std::string gamepadButtonToString(GamepadButton key);
+
+    /// Convert a string to a GamepadButton enum.
```
Comment (RiscadoA) [inline core/include/cubos/core/io/gamepad.hpp:77]:
```suggestion
    /// @brief Convert a string to a GamepadButton enum.
```

File: core/include/cubos/core/io/gamepad.hpp [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    /// @param str The string to convert.
+    /// @return The GamepadButton enum represented by the string.
+    GamepadButton stringToGamepadButton(const std::string& str);
+
+    /// Convert a GamepadAxis enum to a string.
```
Comment (RiscadoA) [inline core/include/cubos/core/io/gamepad.hpp:82]:
```suggestion
    /// @brief Convert a GamepadAxis enum to a string.
```

File: core/include/cubos/core/io/gamepad.hpp [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    /// @param key The GamepadAxis enum to convert.
+    /// @return The string representation of the GamepadAxis enum.
+    std::string gamepadAxisToString(GamepadAxis key);
+
+    /// Convert a string to a GamepadAxis enum.
```
Comment (RiscadoA) [inline core/include/cubos/core/io/gamepad.hpp:87]:
```suggestion
    /// @brief Convert a string to a GamepadAxis enum.
```

File: engine/src/cubos/engine/renderer/plugin.cpp [hunk 6]
```diff
@@ -23,10 +24,11 @@ using cubos::core::io::Window;
 using cubos::core::io::WindowEvent;
 using namespace cubos::engine;
 
-static void init(Write<Renderer> renderer, Read<Window> window, Read<Settings> settings)
+static void init(Write<Renderer> renderer, Read<Window> window, Write<Settings> settings)
 {
     auto& renderDevice = (*window)->renderDevice();
     *renderer = std::make_shared<DeferredRenderer>(renderDevice, (*window)->framebufferSize(), *settings);
+    (*renderer)->pps().addPass<PostProcessingBloom>();
```
Comment (RiscadoA) [inline engine/src/cubos/engine/renderer/plugin.cpp:31]: This should be toggleable with a setting.

**Review by DiogoMendonc-a — APPROVED** [review]

**Pull request merged**